src/components/modals/PluginApiModal.js

```diff
@@ -1,54 +1,57 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useContext } from 'react'
 import PropTypes from 'prop-types'
-import uniqueId from 'lodash/uniqueId'
-import { getWebAdminWindow, getWebAdminDocumentBody } from '_/utils/webadmin-dom'
+import { WebAdminModalContext } from '_/utils/react-modals'
 
 import { Modal } from '@patternfly/react-core'
 
 const PluginApiModal = ({
   children,
   title,
   className,
-  modalId = `pluginApiModal-${uniqueId()}`,
+  id,
   isOpen,
   onClose = () => {},
   ...restForModal
 }) => {
+  const modalContext = useContext(WebAdminModalContext)
+
   useEffect(() => {
     if (isOpen) {
-      const targetWindow = getWebAdminWindow()
       const clonedStyles = []
 
-      if (window !== targetWindow) {
+      if (window !== modalContext.targetWindow) {
```

Comment:
It seems that this condition is checked /executed only once and it's always true. However, if it would be false then we would lose all the CSS (no new added, old ones removed).
Is it possible that `window === modalContext.targetWindow` ?

Reply:
Check the latest updates, the code is in `react-modals.js` now.